drivers/espi/espi_taf_npcx.c

```diff
@@ -0,0 +1,470 @@
+/*
+ * Copyright (c) 2023 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_npcx_espi_taf
+
+#include <soc.h>
+#include <zephyr/drivers/espi.h>
+#include <zephyr/drivers/espi_saf.h>
+#include <zephyr/drivers/flash.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(espi_taf, CONFIG_ESPI_LOG_LEVEL);
+
+static const struct device *const spi_dev = DEVICE_DT_GET(DT_ALIAS(taf_flash));
+
+struct espi_taf_npcx_config {
+	uintptr_t base;
+	uintptr_t mapped_addr;
+	uintptr_t rx_plsz;
+	enum NPCX_ESPI_TAF_ERASE_BLOCK_SIZE erase_sz;
+	enum NPCX_ESPI_TAF_MAX_READ_REQ max_rd_sz;
+};
+
+struct espi_taf_npcx_data {
+	sys_slist_t callbacks;
+};
+
+#define HAL_INSTANCE(dev)						\
+	((struct espi_reg *)((const struct espi_taf_npcx_config *)	\
+	(dev)->config)->base)
+
+#define DT_INST_ERASE_SZ_PROP_OR(inst)					\
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(inst, erase_sz),		\
+		    (_CONCAT(NPCX_ESPI_TAF_ERASE_BLOCK_,		\
+		     DT_INST_STRING_TOKEN(inst, erase_sz))),		\
+		    ((NPCX_ESPI_TAF_ERASE_BLOCK_SIZE_4KB)))
+
+#define DT_INST_MAX_READ_SZ_PROP_OR(inst)				\
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(inst, max_read_sz),		\
+		    (_CONCAT(NPCX_ESPI_TAF_MAX_READ_REQ_,		\
+		     DT_INST_STRING_TOKEN(inst, max_read_sz))),		\
+		    ((NPCX_ESPI_TAF_MAX_READ_REQ_64B)))
```

Comment:
you should not need any of these custom macros since these two props have a default, can you try to drop them and just use `DT_INST_NODE_HAS_PROP`? The devicetree libraries should populate the default for you since it's in the binding and then you don't need to handle it here

Reply:
Done.  
Remove these custom macros and only use `DT_INST_STRING_TOKEN` to get value.  
Because these two properties are required, there is no need to check with `DT_INST_NODE_HAS_PROP`.